src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -3756,6 +3778,57 @@ ErrorIfQueryHasUnroutableModifyingCTE(Query *queryTree)
 }
 
 
+#if PG_VERSION_NUM >= PG_VERSION_14
+
+/*
+ * ErrorIfQueryHasCTEWithSearchClause checks if the query contains any common table
+ * expressions with search clause and errors out if it does.
+ */
+static DeferredErrorMessage *
+ErrorIfQueryHasCTEWithSearchClause(Query *queryTree)
+{
+	if (query_tree_walker(queryTree, ContainsSearchClauseWalker, NULL, 0))
```

Comment:
I don't think recursing only with query_tree_walker is sufficient. See `ExtractQueryWalker()`, we should also recurse into `expression_tree_walker` because some subqueries might be inside expressions (e.g., subqueries in `WHERE` clause)